how-to/openldap/ldap-saslauthd-kerberos.md

```diff
@@ -0,0 +1,282 @@
+(ldap-saslauthd-kerberos)=
+# How to configure OpenLDAP with pass-through SASL authentication using Kerberos
+There exists applications that can only use "simple" username and password [Lightweight Directory Access Protocol](https://ldap.com/) (LDAP) authentication. Pass-though password authentication is a solution to this problem.
+
+## Before you begin
+It is assumed you are starting with a working OpenLDAP server, with a hostname of `ldap-server.example.com`. If not, follow this guide {ref}`Install and configure OpenLDAP<install-openldap>` to set it up. It is also assumed that the `EXAMPLE.COM` realm is set up, and the Kerberos client tools (krb5-user) are installed on the ldap server. You will need to create an ubuntu principal. See {ref}`How to install a Kerberos server <install-a-kerberos-server>`. You should also know how to create service principals. See {ref}`How to configure Kerberos service principals <configure-service-principals>`.
+All the followint configuration will be on `ldap-server.example.com`.
+> **Note**:
+> This process is not the same as using [Generic Security Services Application Programming Interface](https://www.openldap.org/doc/admin26/sasl.html#GSSAPI) (GSSAPI) to log into the LDAP server.
+> Rather it is using [simple authentication](https://www.openldap.org/doc/admin26/security.html#%22simple%22%20method) with the OpenLDAP server so this should be over a [Transport Layer Security](https://datatracker.ietf.org/wg/tls/documents/) (TLS) connection.
+> This process also does not require the Kerberos backend to be OpenLDAP server, the default Keberberos backend if sufficient.
+> The test user we will be using is `ubuntu@EXAMPLE.COM` which must exist in the Kerberos database
+
+
+## Package installation
+Install saslauthd
+
+```bash
+sudo apt install sasl2-bin
+```
+## Saslauthd service principal
+It is assumed you have an admin principal set up in the realm. We will use `ubuntu/admin@EXAMPLE.COM`.
+Next a principal who will be querying Kerberos is needed. The default for saslauthd is "host"/"hostname.domain"@"REALM". For the basic case, it is important that the hostname and domain be what nslookup on your IP number says it is.
+
+## Check the hostname
+Get the hostname from the server
+```bash
+hostname
+```
+Which should give you the hostname of:
+```text
+ldap-server
+```
+Next check what nslookup gives as the full canonical name
+```bash
+nslookup ldap-server
+```
+Which should look something like this.
+```text
+Non-authoritative answer:
+Name:   ldap-server.example.com
+Address: (your IP)
+```
```

Comment:
I suggest replacing the above with a call to `hostname -f`, that will use the same mechanism (append search domain to `ldap-server`, and query DNS).